webapp/channels/src/components/team_settings/team_access_tab/team_access_tab.tsx

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import {RefreshIcon} from '@mattermost/compass-icons/components';
+import type {Team} from '@mattermost/types/teams';
+
+import SelectTextInput, {type SelectTextInputOption} from 'components/common/select_text_input/select_text_input';
+import Input from 'components/widgets/inputs/input/input';
+import BaseSettingItem, {type BaseSettingItemProps} from 'components/widgets/modals/components/base_setting_item';
+import CheckboxSettingItem from 'components/widgets/modals/components/checkbox_setting_item';
+import ModalSection from 'components/widgets/modals/components/modal_section';
+import SaveChangesPanel, {type SaveChangesPanelState} from 'components/widgets/modals/components/save_changes_panel';
+
+import OpenInvite from './open_invite';
+
+import type {PropsFromRedux, OwnProps} from '.';
+
+import './team_access_tab.scss';
+
+type Props = PropsFromRedux & OwnProps;
+
+const AccessTab = (props: Props) => {
+    const generateAllowedDomainOptions = (allowedDomains?: string) => {
+        if (!allowedDomains || allowedDomains.length === 0) {
+            return [];
+        }
+        const domainList = allowedDomains.includes(',') ? allowedDomains.split(',') : [allowedDomains];
+        return domainList.map((domain) => domain.trim());
+    };
+
+    const [inviteId, setInviteId] = useState<Team['invite_id']>(props.team?.invite_id ?? '');
+    const [allowedDomains, setAllowedDomains] = useState<string[]>(generateAllowedDomainOptions(props.team?.allowed_domains));
+    const [showAllowedDomains, setShowAllowedDomains] = useState<boolean>(allowedDomains?.length > 0);
+    const [allowOpenInvite, setAllowOpenInvite] = useState<boolean>(props.team?.allow_open_invite ?? false);
+    const [saveChangesPanelState, setSaveChangesPanelState] = useState<SaveChangesPanelState>('saving');
+    const [inviteIdError, setInviteIdError] = useState<BaseSettingItemProps['error'] | undefined>();
+    const {formatMessage} = useIntl();
+
+    useEffect(() => {
+        setInviteId(props.team?.invite_id || '');
+    }, [props.team?.invite_id]);
```

Comment:
@larkox  This was needed because it seems like when you update any other `invite_id` is also updated. Please see the screen recording below
[Screencast from 08-01-2024 17:44:57.webm](https://github.com/mattermost/mattermost/assets/37421564/33673008-20f0-4b3e-a77b-88ee78686261)